code/modules/reagents/dispenser/dispenser2.dm

```diff
@@ -58,13 +63,11 @@
 	if(user)
 		if(user.unEquip(C))
 			to_chat(user, "<span class='notice'>You add \the [C] to \the [src].</span>")
-		else
-			return
-
-	C.forceMove(src)
-	cartridges[C.label] = C
-	cartridges = sortAssoc(cartridges)
-	SSnano.update_uis(src)
+			C.forceMove(src)
+			cartridges[C.label] = C
+			cartridges = sortTim(cartridges, /proc/cmp_text_asc)
+			SSnano.update_uis(src)
+			return TRUE
```

Comment:
This, on the other hand, doesn't work: it means mapped machines don't get any cartridges on init.